src/main/java/com/microfocus/application/automation/tools/octane/tests/junit/JUnitXmlIterator.java

```diff
@@ -265,11 +266,14 @@ private void handleJUnitTest(XMLEvent event) throws XMLStreamException, IOExcept
                         testReportCreated = optional.isPresent();
                     }
 
-                    //workspace.createTextTempFile("build" + buildId + "." + cleanTestName(testName) + ".", "", "Created  " + testReportCreated);
                     if (testReportCreated) {
                         final String basePath = ((List<String>) additionalContext).get(0);
+                        String nodeName = JenkinsUtils.getCurrentNode(workspace) != null && !JenkinsUtils.getCurrentNode(workspace).getNodeName().isEmpty() ?
```

Comment:
Hi Nissim, if possible I would suggest to declare a variable (i.e. String currentNode = JenkinsUtils.getCurrentNode(workspace) ) and use it instead of calling the getCurrentNode() three times. What do you think?